src/planar/preprocessors.jl

```diff
@@ -68,259 +68,314 @@ _is_adjoint(ex) = isexpr(ex, TO.prime)
+        unchanged = true
+        for i in 1:length(ex.args)
+            newargs[i], unchangeda = _remove_braidingtensors!(args[i], indexmap)
+            unchanged = unchanged && unchangeda
+        end
+        newex = Expr(ex.head, newargs...)
+        return newex, unchanged
+    elseif isexpr(ex, :call) && ex.args[1] == :*
+        args = ex.args
+        newargs = copy(args)
+        unchanged = map(i -> false, args)
+        unchanged[1] = true
+        for i in 2:length(ex.args)
+            newargs[i], unchanged[i] = _remove_braidingtensors!(newargs[i], indexmap)
+        end
+        all(unchanged) && return ex, true
+        while !all(unchanged)
+            for i in 2:length(ex.args)
+                newargs[i], unchanged[i] = _remove_braidingtensors!(newargs[i], indexmap)
             end
         end
-    end
-
-    ex = TO.replaceindices(i -> get(indexmap, i, i), ex)
-    return _purge_braidingtensors(ex)
-end
-
-function _purge_braidingtensors(ex) # actually remove the braidingtensors
-    ex isa Expr || return ex
-    args = collect(filter(ex.args) do a
-                       if isexpr(a, :call) && a.args[1] == :conj
-                           a = a.args[2]
-                       end
-                       if a isa Expr && TO.istensor(a) &&
-                          _remove_adjoint(TO.gettensorobject(a)) == :τ
-                           _, leftind, rightind = TO.decomposetensor(a)
-                           (leftind[1] == rightind[2] && leftind[2] == rightind[1]) ||
-                               throw(ArgumentError("unable to remove braiding tensor $a"))
-                           return false
-                       end
-                       return true
-                   end)
-
-    # multiplication with only a single argument is (rightfully) seen as invalid syntax
-    if isexpr(ex, :call) && args[1] == :* && length(args) == 2
-        return _purge_braidingtensors(args[2])
+        return Expr(ex.head, newargs...), false
+    elseif isexpr(ex, :call) && ex.args[1] ∈ (:+, :-)
+        newargs = copy(ex.args)
+        indexmaps = [copy(indexmap) for _ in 1:(length(newargs) - 1)]
+        unchanged = true
+        for i in 2:length(ex.args)
+            newargs[i], unchangeda = _remove_braidingtensors!(ex.args[i], indexmaps[i - 1])
+            unchanged = unchanged && unchangeda
+        end
+        newex = Expr(ex.head, newargs...)
+        return newex, unchanged
+    elseif isexpr(ex, :call) && ex.args[1] == :/ && length(ex.args) == 3
+        newarg, unchanged = _remove_braidingtensors!(ex.args[2], indexmap)
+        return Expr(:call, :/, newarg, ex.args[3]), unchanged
+    elseif isexpr(ex, :call) && ex.args[1] == :\ && length(ex.args) == 3
+        newarg, unchanged = _remove_braidingtensors!(ex.args[3], indexmap)
+        return Expr(:call, :\, ex.args[2], newarg), unchanged
     else
-        return Expr(ex.head, map(_purge_braidingtensors, args)...)
+        error("unexpected expression $ex")
```

Comment:
in line 547 this is an `ArgumentError`, maybe switch this as well to be consistent?
```suggestion
        throw(ArgumentError("unexpected expression $ex"))
```